dialog/color.go

```diff
@@ -166,16 +167,30 @@ func newColorButtonBox(colors []color.Color, icon fyne.Resource, callback func(c
 	return container.NewGridWithColumns(8, objects...)
 }
 
-func newCheckeredBackground() *canvas.Raster {
-	return canvas.NewRasterWithPixels(func(x, y, _, _ int) color.Color {
-		const boxSize = 8
+const (
+	boxSize       = 8
+	numberOfRings = 12
+)
 
+func newCheckeredBackground(radial ...bool) *canvas.Raster {
+	rect := func(x, y, _, _ int) color.Color {
```

Comment:
it seems wasteful to create `rect` then `f` then re-assign it later if `radial`... If you put an `if radial` at the top then you could simply return this radial version, avoiding both branches running.

Reply:
yes, but both branches need to run for radial anyway and this way the colours for the boxes are only in one place.
what do you think about moving the part that makes the function for `canvas.NewRasterWithPixels` to be another function and having it make a loop of recursion?
something like
```go
func newCheckeredBackground(radial bool) *canvas.Raster {
	return canvas.NewRasterWithPixels(checkeredBackgroundRasterFunction(radial))
}

func checkeredBackgroundRasterFunction(radial bool) func(x, y, w, h int) color.Color {
	if !radial {
		return func(x, y, _, _ int) color.Color {
			if (x/checkeredBoxSize)%2 == (y/checkeredBoxSize)%2 {
				return color.Gray{Y: 58}
			}

			return color.Gray{Y: 84}
		}
	}

	rect := checkeredBackgroundRasterFunction(false)
	return func(x, y, w, h int) color.Color {
		r, t := cmplx.Polar(complex(float64(x)-float64(w)/2, float64(y)-float64(h)/2))
		x = int((t + math.Pi) / (2 * math.Pi) * checkeredNumberOfRings * checkeredBoxSize)
		y = int(r)
		return rect(x, y, 0, 0)
	}
}
```

Reply:
That is very clever, but I think a little confusing to read.
You were probably right to declare the function as `rect` initially, but then it is easier to read if you `if !radial {return new raster etc...}`,  then the rest of the method sets up the radial and you can return the `f` func as an anonymous pass to the raster init.

Though as you point out it is more complex than I realised, so the initial implementation isn't so far from the simplest as I thought, sorry. But in general returning early is a good thing :).